iris_grib/_load_convert.py

```diff
@@ -928,10 +928,16 @@ def _calculate_proj_coords_from_grid_lengths(section, cs):
     mm_to_m = 1e-3
 
     if section['gridDefinitionTemplateNumber'] in _XYGRIDLENGTH_GDT_NUMBERS:
-        dx = section['Dx']
-        dy = section['Dy']
-        nx = section['Nx']
-        ny = section['Ny']
+        if section['gridDefinitionTemplateNumber'] == 140:
+            dx = section['xDirectionGridLengthInMillimetres']
```

Comment:
Did you check that the shorter names don't in fact work?
 -- if not, we wouldn't need this test, or this section because the generic ("else") code can do it.

I ask because I can see that the template definition 3.140 in the official docs (https://community.wmo.int/activity-areas/wmo-codes/manual-codes/volume-i2)  _does_ at least reference the shorter names, 
e.g. "31–34  Nx – number of points along the x-axis".

In the end though, we can only use what implemented in eccodes.

Reply:
as above